### PR TITLE
Trap focus within chat window

### PR DESCRIPTION
**Pull request by sushmi21**
Current behavior: During Tab/ Shift + Tab navigation, focus can be moved to elements outside the chat window. 
Expected behavior: When enableFocusTrap is set to true, focus should never leave the chat window. By default, enableFocusTrap is false, so focus can leave the chat window.

To test: 
- Add webchat setting `enableFocusTrap: true`
- Navigate using keyboard and see is the focus does not leave the chat. The focus order should also be logical during both Tab and Shift + Tab navigation
- Also test by setting enableFocusTrap to false. There should not be any change in the behavior

### CODE REVIEW

File: src/common/interfaces/webchat-config.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ export interface IWebchatSettings {
     enableConnectionStatusIndicator: boolean;
     showEngagementMessagesInChat: boolean;
     enableFileUpload: boolean;
+    enableFocusTrap: boolean;
```
Comment (pedily) [inline src/common/interfaces/webchat-config.ts:22]: you will also need to add an entry to `docs/embedding.md` explaining what this flag does

File: src/webchat-ui/components/WebchatUI.tsx [hunk 2]
```diff
@@ -314,6 +314,47 @@ export class WebchatUI extends React.PureComponent<React.HTMLProps<HTMLDivElemen
             />
         );
     }
+    
+    handleKeydown = (event) => {
+        const closeButtonInHeader = document.getElementById("webchatHeaderCloseButton");
```
Comment (pedily) [inline src/webchat-ui/components/WebchatUI.tsx:319]: Relying on document.getElementById seems kind of "hacky" as we might accidently collide with other elements on the page.

Would it be possible to use `refs` here? Then we would explicitly assign dom element instances within the webchat.

File: src/webchat-ui/components/WebchatUI.tsx [hunk 3]
```diff
@@ -368,6 +409,7 @@ export class WebchatUI extends React.PureComponent<React.HTMLProps<HTMLDivElemen
 							className="webchat-root"
 							aria-labelledby="webchatHeaderTitle"
 							role="region"
+							onKeyDown={enableFocusTrap && open ? this.handleKeydown : () => {}}
```
Comment (pedily) [inline src/webchat-ui/components/WebchatUI.tsx:412]: I'd move the `open` check to `handleKeyDown` and always trigger it.

File: src/webchat-ui/components/WebchatUI.tsx [hunk 4]
```diff
@@ -314,6 +314,47 @@ export class WebchatUI extends React.PureComponent<React.HTMLProps<HTMLDivElemen
...
+                } else if (getStartedButton) {
+                    getStartedButton.focus();
+                } else if (webchatInputButtonMenu) {
+                    webchatInputButtonMenu.focus();
+                } else {
+                    chatToggleButton?.focus()
+                }
+            }
+        }
+        if(event.target === chatToggleButton) {
```
Comment (pedily) [inline src/webchat-ui/components/WebchatUI.tsx:340]: These cases deserve a line of comment or two. What is special about pressing (shift)+tab on these two elements?

**Review by pedily — CHANGES_REQUESTED** [review]

File: src/webchat-ui/components/WebchatUI.tsx (hunk 4 above)
Comment (sushmi21) [reply src/webchat-ui/components/WebchatUI.tsx]: Added comments

File: src/webchat-ui/components/WebchatUI.tsx (hunk 3 above)
Comment (sushmi21) [reply src/webchat-ui/components/WebchatUI.tsx]: Fixed

File: src/common/interfaces/webchat-config.ts (hunk 1 above)
Comment (sushmi21) [reply src/common/interfaces/webchat-config.ts:22]: Fixed

**Review by kwinto — APPROVED** [review]
Works as expected

File: src/webchat-ui/components/WebchatUI.tsx (hunk 2 above)
Comment (sushmi21) [reply src/webchat-ui/components/WebchatUI.tsx]: Replaced ids with refs wherever possible.

**Review by dshire — APPROVED** [review]
working as expected

**Pull request merged**